indico/modules/events/management/forms.py

```diff
@@ -230,6 +230,17 @@ def __init__(self, *args, **kwargs):
             del self.label_message
 
 
+class PrivacyDashboardForm(IndicoForm):
+    _data_controller_fields = ('data_controller_name', 'data_controller_email')
+    _privacy_policy_fields = ('privacy_policy_url', 'privacy_policy')
+    data_controller_name = StringField(_('Person/Institution'))
+    data_controller_email = EmailField(_('Contact email'), [Optional(), Email()])
+    privacy_policy_url = URLField(_('URL'), [Optional(), URL()],
+                                  description=_('The URL to an external page with the privacy policy'))
+    privacy_policy = TextAreaField(_('Text'), widget=CKEditorWidget(),
+                                   description=_('Only used if no URL is provided'))
+
```

Comment:
Some validation logic that disallows providing both a url and text would be useful. In the admin UI it's not so important but I'm sure we have some event managers that don't realize the fields are mutually exclusive..

Something like this should work (untested):

```python
    def validate_privacy_policy(self, field):
        if self.privacy_policy_url.data and self.privacy_policy.data:
            raise ValidationError(_('...'))

    privacy_policy_url = privacy_policy
```

Reply:
Adding to this, and keeping in mind the suggested change by Adrian might be so you can define an explicit error message. But maybe it would be worth considering using the `Exclusive` validator instead of custom logic:

```python
    privacy_policy_url = URLField(_('URL'), [Optional(), URL(), Exclusive('privacy_policy')],
                                  description=_('The URL to an external page with the privacy notice'))
    privacy_policy = TextAreaField(_('Text'), [Exclusive('privacy_policy_url')], widget=CKEditorWidget(),
                                   description=_('Only used if no URL is provided'))
```